### PR TITLE
Add health checks routing extensions

### PR DESCRIPTION
**Pull request by JamesNK**
TODO:

- [x] ~Support port?~ done later via a matcher policy https://github.com/aspnet/AspNetCore/issues/5957
- [x] Docs

Fixes https://github.com/aspnet/AspNetCore/issues/4617

### CODE REVIEW

**Comment by JamesNK** [comment]
@rynowak UseHealthChecks supports limiting access to a port. What do you think of making the `DomainMatcherPolicy` that I wrote ship in routing? Then it would be a simple matter of setting the appropriate metadata in for the healthchecks endpoint.

File: src/Middleware/HealthChecks/src/Builder/HealthCheckEndpointRouteBuilderExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.Diagnostics.HealthChecks.Builder
```
Comment (rynowak) [inline src/Middleware/HealthChecks/src/Builder/HealthCheckEndpointRouteBuilderExtensions.cs]: This should be `Microsoft.AspNetCore.Builder`

File: src/Middleware/HealthChecks/src/Builder/HealthCheckEndpointRouteBuilderExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+{
+    public static class HealthCheckEndpointRouteBuilderExtensions
```
Comment (rynowak) [inline src/Middleware/HealthChecks/src/Builder/HealthCheckEndpointRouteBuilderExtensions.cs]: docccs

**Comment by rynowak** [comment]
As discussed we want support for filtering on host/port, I'm ok with that to land separately.

**Review by rynowak — CHANGES_REQUESTED** [review]
Seems good. Need to get final polish

**Comment by JamesNK** [comment]
re: host/port

UseMiddleware has port as an argument. Do you have a strong preference on whether MapHealthChecks should have an overload that takes a port and adds the metadata for you, or do we tell people to use a standard extension method to add port metadata.

I lean towards and extension method but I don't have a strong preference either way.

**Comment by rynowak** [comment]
I lean towards extension method as well if it's easy to use. If not we can add it back as first-class, so lets start with the extension method

**Review by rynowak — APPROVED** [review]

**Pull request merged**